meta.go

```diff
@@ -56,74 +56,86 @@ func getMeta(key string, metaSpace string, output io.Writer) error {
 	return nil
 }
 
+// indexOfRightBracket gets index of right bracket("]"). e.g. the key is foo[10], return 6
+func indexOfRightBracket(key string, current int) int {
+	var i int
+	for i = current + 1; ; i++ {
+		_, err := strconv.Atoi(string(key[i])) // Check the next char is integer
```

Comment:
Just wondering if it would be more clear to use the `regexp` package to do something like this
```go
package main

import (
	"fmt"
	"regexp"
)

func main() {
	re := regexp.MustCompile(`\[(.*?)\]`)
	s := `foo[10].bar[4].baz`
	fmt.Println((re.FindStringIndex(s))[1] - 1)
}

// prints 6
```